Branch-SDK-TestBed/src/io/branch/branchandroiddemo/MainActivity.java

```diff
@@ -244,60 +237,56 @@ public void onClick(View v) {
-                        .setStage("stage1")
-                        .setFeature("feature1")
-                        .addTag("Tag1")
-                        .addTag("Tag2")
-                        .setDefaultURL("https://play.google.com/store/apps/details?id=com.kindred.android")
-                        .setCallback(new Branch.BranchLinkShareListener() {
-                            @Override
-                            public void onShareLinkDialogLaunched() {
-                                Log.i("BranchTestBed", "onShareLinkDialogLaunched()");
-                            }
-
-                            @Override
-                            public void onShareLinkDialogDismissed() {
-                                Log.i("BranchTestBed", "onShareLinkDialogDismissed()");
-                            }
-
-                            @Override
-                            public void onLinkShareResponse(String sharedLink, String sharedChannel, BranchError error) {
-                                if (error != null) {
-                                    Log.i("BranchTestBed", "onLinkShareResponse... " + sharedLink + " " + sharedChannel + " " + error.getMessage());
-                                } else {
-                                    Log.i("BranchTestBed", "onLinkShareResponse... " + sharedLink + " " + sharedChannel);
-                                }
-                            }
+                        .addPreferredSharingOption(SharingHelper.SHARE_WITH.TWITTER);
 
-                            @Override
-                            public void onChannelSelected(String channelName) {
-                                Log.i("BranchTestBed", "onChannelSelected... " + channelName);
-                            }
-                        })
-                        // Custom style for Copy url and More options
-                        //.setCopyUrlStyle(getResources().getDrawable(android.R.drawable.ic_menu_send),"Save this URl","Link added to clipboard")
-                        //.setMoreOptionStyle(getResources().getDrawable(android.R.drawable.ic_menu_search), "Show more")
-                        .shareLink();
+                branchUniversalObject.showShareSheet(MainActivity.this, linkProperties, shareSheetStyle, new Branch.BranchLinkShareListener() {
+                    @Override
+                    public void onShareLinkDialogLaunched() {
+                    }
+                    @Override
+                    public void onShareLinkDialogDismissed() {
+                    }
+                    @Override
+                    public void onLinkShareResponse(String sharedLink, String sharedChannel, BranchError error) {
+                    }
+                    @Override
+                    public void onChannelSelected(String channelName) {
+                    }
+                });
             }
         });
+
+        branchUniversalObject = new BranchUniversalObject("canonical/identifier/", "My Content Title")
```

Comment:
Yes .  The BranchUniversalObject is created before calling the share sheet. (`BranchUniversalObject` is created on `onCreate()` and the Share sheet is called on a button click event.  Probably I should move to to the first statement in `onCreate()` to avoid a confusion